ndc-test/src/snapshot.rs

```diff
@@ -61,17 +61,21 @@ impl<'a, C: Connector> Connector for SnapshottingConnector<'a, C> {
 
 pub fn snapshot_test<R>(snapshot_path: &Path, expected: &R) -> Result<()>
 where
-    R: serde::Serialize + serde::de::DeserializeOwned + PartialEq,
+    R: serde::Serialize + serde::de::DeserializeOwned + PartialEq + std::fmt::Debug,
 {
     if snapshot_path.exists() {
         let snapshot_file = File::open(snapshot_path).map_err(Error::CannotOpenSnapshotFile)?;
         let snapshot: R = serde_json::from_reader(snapshot_file)?;
+        let compare: pretty_assertions::Comparison<R, R> =
+            pretty_assertions::Comparison::new(&snapshot, expected);
 
         if snapshot != *expected {
             let actual = serde_json::to_string_pretty(&expected)?;
+
             return Err(Error::ResponseDidNotMatchSnapshot(
                 snapshot_path.to_path_buf(),
                 actual,
+                format!("{compare}"),
```

Comment:
Is there any use in storing this here, or can we just move the `format!` into the `Display` implementation for the error type?

Reply:
So I did want to move into the error, but then compare's type is `pretty_assertions::Comparison<R, R>` and it would require extending parameterizing the `Error` type to `Error<R>`, which would be a substantial change, so for simplicity, I am converting it to String.